### PR TITLE
HMSPROV-181: Add dao test for instance_reservation

### PR DESCRIPTION
**Pull request by adiabramovitch**
- [x]  TestCreateInstance
- [x] TestUpdateReservationIDForAWS

### CODE REVIEW

**Review by avitova — COMMENTED** [review]
What should happen if there is no Reservation with ReservationID of the new instance? Do we need to handle those cases? I was just thinking we might want to add a test like this, just to make sure that CreateInstance acts the way we need.

File: internal/models/reservation_model.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ type AWSReservation struct {
 
 	// The ID of the image from which the instance is created.
 	AMI string `db:"ami" json:"ami"`
+
+	// The ID of the the aws reservation which was created.
```
[the remark this answers is not in the document]
Comment (adiabramovitch) [reply internal/models/reservation_model.go:56]: It must have a ReservationID or else the instance wouldn't have been created
I mean: We first create the reservation, and only after the reservation was created successfully we will launch an aws instance and create a db record

[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply internal/models/reservation_model.go:56]: I think it make sense, we can't test every corner case, so unless there is a real worry it could cause some real trouble in the future, I'm ok not to have explicit test for that :+1:

[the remark this answers is not in the document]
Comment (ezr-ondrej) [reply internal/models/reservation_model.go:56]: @adiabramovitch can you fix the nit? :)

**Review by ezr-ondrej — APPROVED** [review]
Apart of THE nit, I'm good here :+1:
